specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.2/TextAnalytics.json

```diff
@@ -0,0 +1,1481 @@
+              "$ref":"#/definitions/DocumentStatistics"
+           }
+        }
+     },
+     "DetectedLanguage":{
+        "type":"object",
+        "required":[
+           "name",
+           "iso6391Name",
+           "confidenceScore"
+        ],
+        "properties":{
+           "name":{
+              "type":"string",
+              "description":"Long name of a detected language (e.g. English, French)."
+           },
+           "iso6391Name":{
+              "type":"string",
+              "description":"A two letter representation of the detected language according to the ISO 639-1 standard (e.g. en, fr)."
+           },
+           "confidenceScore":{
+              "type":"number",
+              "format":"double",
+              "description":"A confidence score between 0 and 1. Scores close to 1 indicate 100% certainty that the identified language is true."
+           }
+        }
+     }
+  },
+  "parameters":{
+     "Endpoint":{
+        "name":"Endpoint",
+        "description":"Supported Cognitive Services endpoints (protocol and hostname, for example: https://westus.api.cognitive.microsoft.com).",
+        "x-ms-parameter-location":"client",
+        "required":true,
+        "type":"string",
+        "in":"path",
+        "x-ms-skip-url-encoding":true
+     },
+     "StringIndexType":{
+        "name":"stringIndexType",
+        "in":"query",
+        "type":"string",
+        "description":"(Optional) Specifies the method used to interpret string offsets.  Defaults to Text Elements (Graphemes) according to Unicode v8.0.0. For additional information see https://aka.ms/text-analytics-offsets",
+        "default":"TextElements_v8",
+        "enum":[
+           "TextElements_v8",
+           "UnicodeCodePoint",
+           "Utf16CodeUnit"
+        ],
+        "x-ms-enum":{
+           "name":"StringIndexType",
+           "modelAsString":false,
+           "values":[
+              {
+                 "value":"TextElements_v8",
+                 "description":"Returned offset and length values will correspond to TextElements (Graphemes and Grapheme clusters) confirming to the Unicode 8.0.0 standard. Use this option if your application is written in .Net Framework or .Net Core and you will be using StringInfo."
+              },
+              {
+                 "value":"UnicodeCodePoint",
+                 "description":"Returned offset and length values will correspond to Unicode code points. Use this option if your application is written in a language that support Unicode, for example Python."
+              },
+              {
+                 "value":"Utf16CodeUnit",
+                 "description":"Returned offset and length values will correspond to UTF-16 code units. Use this option if your application is written in a language that support Unicode, for example Java, JavaScript."
+              }
+           ]
+        },
+        "x-ms-parameter-location":"client"
```

Comment:
Why are you choosing to make it a client parameter instead of a method parameter?

Reply:
For python multiapi, it'll be better to have this on the method level. We haven't dealt with client parameters that are required for some versions, and don't exist for others

Reply:
@assafi had this defined in another PR which I just copied over. Assaf is there a reason to have it like this? @thediris if you have an answer for this please let us know

Reply:
@johanste do you have any input regarding string index type being a client parameter?

Reply:
I believe this was based on the assumption that we wouldn't provide a parameter at all for the encoding. But based on the conversation in the arch board review (scenario where I may want to change it when storing something in a database to be consumed by a different language etc.), it is probably better to (still not exposing it as a parameter in the first release) have it as a per-method parameter.

Reply:
I have changed it to method. @assafi Please let us know if you have any concerns

Reply:
Yep. Agreed with everything that's been said. Thanks for the feedback everyone.